### PR TITLE
Fix warnings

### LINKED ISSUE #97: Fix build warnings
Preparational step for releasing CENNZnet.

### PR DESCRIPTION
**Pull request by aliXsed**
Closes #97

### CODE REVIEW

File: frame/balances/src/mock.rs [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 //! Test utilities
 
+// TODO remove the following "allow" when merging in the latest version from upstream
+#![allow(dead_code)]
```
Comment (jordy25519) [inline frame/balances/src/mock.rs:20]: Can't say I'm a big fan of leaving TODOs like this.
What is "latest version" everytime or next time only?

**Review by jordy25519 — COMMENTED** [review]
Not sure `#![allow(dead_code)]` is the right thing. Would prefer add a comment why some stuff is dead code or remove the dead code instead

**Comment by aliXsed** [comment]
> Not sure `#![allow(dead_code)]` is the right thing. Would prefer add a comment why some stuff is dead code or remove the dead code instead

I have done that in places where there were several dead functions in the file with no signs of the code being removed in the upstream. So in those cases, I preferred to leave a very minimal footprint in the file to keep future upstream merges easy.

File: frame/balances/src/mock.rs (hunk 1 above)
Comment (aliXsed) [reply frame/balances/src/mock.rs:20]: I have assumed the upstream warnings to be transitional. Anyone doing the next upstream merge would definitely come across this line and should make a decision about it. Ideally, with the next merge, we could remove the "allow" and that would be the preferred choice. Perhaps a better comment is needed here. One good practice with TODO comments that I know is there should be an issue created in the backlog and a reference to the issue number in the TODO comment, what's your opinion about this?

**Review by jordy25519 — APPROVED** [review]

**Pull request merged**